flux-middle/src/rustc/ty.rs

```diff
@@ -157,14 +159,39 @@ pub enum AliasKind {
 }
 
 #[derive(Clone, PartialEq, Eq, Hash, Encodable, Decodable)]
-pub struct Const {
+pub struct ValueConst {
     pub val: usize,
 }
 
+#[derive(Clone, PartialEq, Eq, Hash, Encodable, Decodable)]
+pub struct ParamConst {
+    pub index: u32,
+    pub name: Symbol,
+}
+
+#[derive(Clone, PartialEq, Eq, Hash, Encodable, Decodable)]
+pub enum Const {
+    Param(ParamConst),
+    Value(ValueConst),
+}
+
+impl From<&ArrayLen> for Const {
+    fn from(len: &ArrayLen) -> Self {
+        Const::Value(ValueConst { val: len.val })
+    }
+}
```

Comment:
Let's remover this to avoid the dependency on `fhir`.

```suggestion
```

Reply:
done.